core/arch/arm/kernel/boot.c

```diff
@@ -1082,10 +1088,21 @@ void __weak boot_init_primary_final(void)
 	if (!IS_ENABLED(CFG_WITH_PAGER))
 		boot_mem_release_tmp_alloc();
 
-	if (!IS_ENABLED(CFG_NS_VIRTUALIZATION))
+	if (!IS_ENABLED(CFG_NS_VIRTUALIZATION)) {
+		/* Unmask native interrupts during init/finalcalls */
+		thread_set_exceptions(thread_get_exceptions() &
+				      ~THREAD_EXCP_NATIVE_INTR);
+
 		call_driver_initcalls();
+	}
+
 	call_finalcalls();
```

Comment:
`CFG_NS_VIRTUALIZATION` may also benefit from unmasked native interrupts during finalcalls.

Reply:
Right, I missed that. Thanks, i'll update.